src/utilities/transition/ExpandableTransition.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+// Required modules.
+var Events = require( '../../mixins/Events.js' );
+var BaseTransition = require( './BaseTransition' );
+var fnBind = require( '../function-bind' ).bind;
+var contains = require( '../dom-class-list' ).contains;
+
+// Exported constants.
+var CLASSES = {
+  BASE_CLASS:     'u-expandable-transition',
+  EXPANDED:       'u-expandable-expanded',
+  COLLAPSED:      'u-expandable-collapsed',
+  OPEN_DEFAULT:   'u-expandable-content__onload-open'
+};
+
+/**
+ * ExpandableTransition
+ * @class
+ *
+ * @classdesc Initializes new ExpandableTransition behavior.
+ *
+ * @param {HTMLNode} element
+ *   DOM element to apply move transition to.
+ * @returns {ExpandableTransition} An instance.
+ */
+function ExpandableTransition( element, CLASSES ) { // eslint-disable-line max-statements, no-inline-comments, max-len
+
+  var _baseTransition = new BaseTransition( element, CLASSES ),
+      timer,
+      isAnimating = false;
+
+  /**
+   * @returns {ExpandableTransition} An instance.
+   */
+  function init() {
+    _baseTransition.init();
+    var _transitionCompleteBinded = fnBind( _transitionComplete, this );
+    _baseTransition.addEventListener( BaseTransition.END_EVENT,
+                                      _transitionCompleteBinded );
+
+    if ( !contains( element, CLASSES.OPEN_DEFAULT ) ) {
+      _baseTransition.applyClass( CLASSES.COLLAPSED );
+    }
+
+    return this;
+  }
+
+  /**
+   * Handle the end of a transition.
+   */
+  function _transitionComplete() {
+    this.trigger( BaseTransition.END_EVENT, { target: this } );
+  }
+
+  /**
+   * Toggle the expandable
+   * @returns {ExpandableTransition} An instance
+   */
+  function toggleExpandable() {
+    if ( !contains( element, CLASSES.COLLAPSED ) ) {
+      collapse();
+    } else {
+      expand();
+    }
+
+    return this;
+  }
+
+  /**
+   * Collapses the expandable content
+   * @returns {ExpandableTransition} An instance.
+   */
+  function collapse() {
+    element.style.maxHeight = '0px';
+    _baseTransition.applyClass( CLASSES.COLLAPSED );
+
+    return this;
+  }
+
+  /**
+   * Expands the expandable content
+   * @returns {ExpandableTransition} An instance.
+   */
+  function expand() {
+    element.style.maxHeight = element.scrollHeight + 'px';
+    _baseTransition.applyClass( CLASSES.EXPANDED );
+
+    return this;
+  }
+
+  // Attach public events.
+  this.addEventListener = Events.on;
```

Comment:
Why not follow the native EventTarget API naming here? https://developer.mozilla.org/en-US/docs/Web/API/EventTarget